### PR TITLE
Feat: Add project property to workflow run to return workspace and project

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem

With workflowRun object obtained using by_id method, there was no way for the user to determine the workspace and project: https://zapatacomputing.atlassian.net/browse/ORQSDK-817

# This PR's solution

Add .project property to WorkflowRun class

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: src/orquestra/sdk/_base/abc.py [hunk 1]
```diff
@@ -194,6 +194,12 @@ def get_workflow_logs(
         """
         raise NotImplementedError()
 
+    def get_workflow_project(self, wf_run_id: WorkflowRunId):
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/abc.py:197]:
```suggestion
    @abstractmethod
    def get_workflow_project(self, wf_run_id: WorkflowRunId) -> ProjectRef:
```

File: src/orquestra/sdk/_base/abc.py [hunk 2]
```diff
@@ -194,6 +194,12 @@ def get_workflow_logs(
...
+        """
+        Returns project and workspace IDs of given workflow
+        """
+        raise WorkspacesNotSupportedError()
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/abc.py:201]: **suggestion**: please don't rely on inheriting the method implementation, this ABC was meant only for specifying the interface to adhere to.
```suggestion
        raise NotImplementedError()
```

File: src/orquestra/sdk/_base/abc.py (hunk 1 above)
Comment (alexjuda) [reply src/orquestra/sdk/_base/abc.py:197]: Looks like we forgot to sprinkle `@abstractmethod` on other methods in this interface too.

**Review by alexjuda — COMMENTED** [review]
I know it's a draft but I got a review request so I left some comments!

**Comment by SebastianMorawiec** [comment]
Ill add changelog entry after new changelog is rolled out

File: tests/sdk/v2/api/test_wf_run.py [hunk 3]
```diff
@@ -588,6 +590,18 @@ def test_error_from_runtime(exc):
                 # When
                 run.stop()
 
+    class TestProject:
+        def test_get_project(self, run):
+            project = run.project
+            assert project.workspace_id == "ws"
+            assert project.project_id == "proj"
+
+        def test_value_get_cashed(self, run):
```
Comment (jamesclark-Zapata) [inline tests/sdk/v2/api/test_wf_run.py:599]:
```suggestion
        def test_value_get_cached(self, run):
```

**Review by jamesclark-Zapata — APPROVED** [review]

**Pull request merged**